ops/docker/Dockerfile.packages

```diff
@@ -0,0 +1,70 @@
+# This Dockerfile builds all the dependencies needed by the monorepo, and should
+# be used to build any of the follow-on services
+#
+# ### BASE: Install deps
+# We do not use Alpine because there's a regression causing it to be very slow
+# when used with typescript/hardhat: https://github.com/nomiclabs/hardhat/issues/1219
+FROM node:14.18.1-buster-slim as base
+
+RUN apt-get update -y && apt-get install -y git
+
+ARG VERSION=v0.8.9
+ARG SOLC_VERSION=${VERSION}+commit.e5eed63a
+ARG SOLC_UPSTREAM=https://github.com/ethereum/solc-bin/raw/gh-pages/linux-amd64/solc-linux-amd64-${SOLC_VERSION}
+
+ADD $SOLC_UPSTREAM ./solc
+ADD https://github.com/ethereum/solc-bin/raw/gh-pages/linux-amd64/solc-linux-amd64-v0.5.17+commit.d19bba13 ./solc
+
+# copy over the needed configs to run the dep installation
+# note: this approach can be a bit unhandy to maintain, but it allows
+# us to cache the installation steps
+WORKDIR /optimism
+COPY *.json yarn.lock ./
+COPY packages/core-utils/package.json ./packages/core-utils/package.json
+COPY packages/common-ts/package.json ./packages/common-ts/package.json
+COPY packages/contracts/package.json ./packages/contracts/package.json
+COPY packages/data-transport-layer/package.json ./packages/data-transport-layer/package.json
+COPY packages/batch-submitter/package.json ./packages/batch-submitter/package.json
+COPY packages/message-relayer/package.json ./packages/message-relayer/package.json
+COPY packages/replica-healthcheck/package.json ./packages/replica-healthcheck/package.json
+COPY packages/regenesis-surgery/package.json ./packages/regenesis-surgery/package.json
+COPY integration-tests/package.json ./integration-tests/package.json
+
+RUN yarn install --frozen-lockfile
+
+COPY ./packages ./packages
+COPY ./integration-tests ./integration-tests
+
+# build it!
+RUN yarn build
+
+# build integration tests' contracts
+RUN yarn workspace @eth-optimism/integration-tests build
+
+
+FROM base as deployer
+RUN apt-get install -y curl jq python3
```

Comment:
We could probably just install `curl jq python3` in the base before

Reply:
Yep good idea.